src/Uno.UI/DirectUI/JoltCollections.h.cs

```diff
@@ -1586,15 +1587,47 @@ private void ClearHandlers()
 	//}
 	#endregion
 
-	internal class TrackerCollection<T> : List<T>, IVector<T>
+	internal class TrackerCollection<T> : IList<T>, IVector<T>
 	{
-		public void RemoveAtEnd()
+		private readonly List<T> _collection = new();
+
+		public virtual T this[int index]
+		{
+			get => _collection[index];
+			set => _collection[index] = value;
+		}
+
+		public int Count => _collection.Count;
+
+		public bool IsReadOnly => false;
+
+		public virtual void Add(T item) => _collection.Add(item);
+
+		public virtual void Insert(int index, T item) => _collection.Insert(index, item);
+
+		public virtual void Clear() => _collection.Clear();
+
+		public bool Contains(T item) => _collection.Contains(item);
+
+		public void CopyTo(T[] array, int arrayIndex) => _collection.CopyTo(array, arrayIndex);
+
+		public IEnumerator<T> GetEnumerator() => _collection.GetEnumerator();
```

Comment:
Maybe better to make the return type `List<T>.Enumerator` instead of `IEnumerator<T>`?

Reply:
The type implements `IList` so I think that is not possible (unless I misunderstood)

Reply:
I think if there is explicit interface implementation, it's valid to change this one

Reply:
There is not explicit implementation, if I am not missing something 🤔 . The explicit one is for `IEnumerator`, but this one is still needed for `IEnumerator<T>` to satisfy `IList<T>` I think